docs/api/loaders.rst

```diff
@@ -261,6 +261,35 @@ This is a completely functional loader for the ``sas7bdat`` (SAS dataset file) f
             with self.dat as fp:
                 yield from Progress(fp, total=self.dat.properties.row_count)
 
+
+Guessing Filetypes
+==================
+
+When loading a file, VisiData tries to infer its filetype by peeking at the initial lines of the file and guessing from its structure.
+
+``vd.guess_<filetype>(path)`` contains this logic for checking whether a file might be ``<filetype``.
+
+If those structures are not present, the function should return nothing. If they are, the function should return a dictionary with:
+
+* ``filetype`` being the filetype they detect (corresponding to the ``vd.open_<filetype>``)
+* ``_likelihood`` (optional) being a number from 0-10, 10 being most likely and 0 meaning a last ditch effort if nothing else will take it
+
+`Examples of guess_filetype functions <https://github.com/saulpw/visidata/commit/4743f92bb855cf931d896e65845c549ce6027e2f>`
+
+
+::
+
+        @VisiData.api
+        def guess_foo(vd, p):
+            line = foolib.iterfoo(p.open_text())
+            try:
```

Comment:
I'm not sure if this `try` is needed anymore.

Reply:
I was building off of an earlier `FooSheet` example! I am not attached. I think the `r[0] = "#Foo"` is demonstrated enough (along with the link to actual examples).